examples/prefix_sum.cpp

```diff
@@ -0,0 +1,156 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-25, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+#include <iomanip>  
+#include <vector>
+#include <cassert>
+#include <chrono>
+
+#include "RAJA/RAJA.hpp"
+#include "memoryManager.hpp"
+
+/*
+ *  Prefix Sum Example
+ *
+ *  Computes an exclusive prefix sum (scan) on an array of integers using
+ *  multiple execution policies.
+ *
+ *  RAJA features shown:
+ *    - `exclusive_scan` operation with different execution backends
+ *    - Use of `make_span` to create RAJA-compatible views
+ *    - Sequential, OpenMP, and CUDA execution variants
+ *    - CUDA device memory allocation and transfer
+ *
+ *  If CUDA is enabled, device memory is allocated manually with `cudaMalloc`
+ *  and the results are copied back to host memory for validation.
+ */
+
+
+/*
+  CUDA_BLOCK_SIZE - specifies the number of threads in a CUDA thread block
+*/
+#if defined(RAJA_ENABLE_CUDA)
+const int CUDA_BLOCK_SIZE = 256;
+#endif
+
+/*
+  N - the length of the series to perform the prefix sum
+*/
+const int N = 100;
+
+bool check_equal(const std::vector<int>& a, const std::vector<int>& b)
+{
+  if (a.size() != b.size()) return false;
+  for (size_t i = 0; i < a.size(); ++i) {
+    if (a[i] != b[i]) return false;
+  }
+  return true;
+}
+
+int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv))
+{
+  std::cout << "\n\nRAJA prefix sum (exclusive_scan) example using a series of " << N << " length...\n";
+
+  std::vector<int> input(N, 1);
+  std::vector<int> reference_output(N, 0);
+  std::vector<int> test_output(N, 0);
+
+  //----------------------------------------------------------------------------
+  std::cout << "\n Running C-style prefix sum...\n";
+
+  auto start = std::chrono::high_resolution_clock::now();

```

Comment:
There is a RAJA timer that might be useful here. See example: 
https://github.com/LLNL/RAJA/blob/develop/exercises/permuted-layout-batch-matrix-multiply_solution.cpp#L105